<launch>

    <arg name="markerSize"      default="0.16"/>    <!-- in m -->
    <arg name="eye"             default="left"/>
    <arg name="board_config"    default="$(find aruco_ros)/samples/multi_board.yml" />
    <arg name="board_dir"       default="$(find aruco_ros)/samples/" />
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->

    <node pkg="aruco_ros" type="multi_board" name="aruco_multi_board">
        <remap from="/camera_info" to="/stereo/$(arg eye)/camera_info" />
        <remap from="/image" to="/stereo/$(arg eye)/image_rect_color" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="stereo_gazebo_$(arg eye)_camera_optical_frame"/>
        <param name="board_config"       value="$(arg board_config)" />
        <param name="board_dir"          value="$(arg board_dir)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
        <param name="draw_markers"       value="True" />
    </node>

</launch>
